contrib/CMakePresets.json

```diff
@@ -0,0 +1,112 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
+      "displayName": "base preset",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "environment": {
+        "LOG_DIR": "${sourceDir}/build/${presetName}/logs"
+      },
+      "hidden": true
+    },
+    {
+      "name": "release",
+      "displayName": "Build as release mode.",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "debug",
+      "displayName": "Build as debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "debug-clang",
+      "displayName": "Build with clang toolchain",
+      "environment": {
+        "CC": "clang",
+        "CXX": "clang++"
+      },
+      "inherits": [
+        "debug"
+      ]
+    },
+    {
+      "name": "asan",
+      "displayName": "Build with ASAN support",
+      "cacheVariables": {
+        "CLANG_ASAN_UBSAN": "ON"
+      },
+      "environment": {
+        "ASAN_OPTIONS": "'detect_leaks=0;log_path=$env{LOG_DIR}'",
+        "UBSAN_OPTIONS": "'print_stacktrace=1'"
```

Comment:
A sign that this is a good change, is that it will allow us to simplify our docs. We can remove some fiddly code from `src/nvim/README.md`. And this is cross-platform (no shell script).

Reply:
I had that in mind, I'm even planning to add support to tests presets (includes controlling `busted`, `valgrind`, `gdb`, etc) once #18674 has landed in some form.